File: inf_bez/1_laba/laba1_f_bez/laba1/test_lab1.py
import unittest
from lab1 import decrypt
from lab1 import crypt
import random


class MyTestCase(unittest.TestCase):

    def test_crypto(self):
        """
        Test that it crypt and decrypt text
        """
        text = "Lorem ipsum dolor sit amet consectetur adipiscing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua Ut enim ad minim veniam quis nostrud exercitation ullamco laboris nisi ".lower()
        _text = crypt(text)
        result = decrypt(_text)
        print("Test 1")
        print("======")
        print("original:  " + text)
        print("crypt   :  " + _text)
        print("decrypt :  " + result)
        self.assertEqual(text, result)

    def test_crypto_enhanced(self):
        """
        Test that it enhanced crypt and decrypt text
        """
        text = "Lorem ipsum dolor sit amet consectetur adipiscing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua Ut enim ad minim veniam quis nostrud exercitation ullamco laboris nisi ".lower()
        rez = True
        _text = crypt(text)
        d = []
        print("Test 2")
        print("======")
        n=0;
        for i in range(10):
            ind = int(random.random() * len(text))
            if text[ind] == ' ':
                continue
            d.append(ord(text[ind]) - ord(_text[ind]))
            if n > 0 :
                rez = rez and d[n] == d[n - 1]
            n = n + 1
        print(d)
        self.assertTrue(not rez)


if __name__ == '__main__':
    unittest.main()

File: inf_bez/1_laba/laba1_f_bez/laba1/test_lab1.java
import static org.junit.jupiter.api.Assertions.*;

import org.junit.jupiter.api.Test;

class test_lab1 {

	@Test
	void test_crypto() {
		lab1 l=new lab1();
		String text="Lorem ipsum dolor sit amet consectetur adipiscing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua Ut enim ad minim veniam quis nostrud exercitation ullamco laboris nisi ".toLowerCase();
		String _text= l.crypt(text);
		String result=l.decrypt(_text);
		System.out.println("test 1");
		System.out.println("	original:  "+text);
		System.out.println("	crypt   :  "+_text);
		System.out.println("	decrypt :  "+result);	
		assertEquals(text, result);
	}

	@Test
	void test_crypto_enhanced() {
		lab1 l=new lab1();
		boolean rez=true;
		String text="Lorem ipsum dolor sit amet consectetur adipiscing elit sed do eiusmod tempor incididunt ut labore et dolore magna aliqua Ut enim ad minim veniam quis nostrud exercitation ullamco laboris nisi ".toLowerCase();
		String _text= l.crypt(text);
		System.out.println("test 2");
		int d[]=new int[10];
		for(int i=0;i<d.length;i++) {
			int ind=(int)(Math.random()*text.length());
			if(text.charAt(ind)==' ') continue;
			d[i]=text.charAt(ind)-_text.charAt(ind);
			System.out.print(text.charAt(ind)-_text.charAt(ind) + " ");
			if(i>0) rez = rez && d[i]==d[i-1]; 
		}
		assert(!rez);
	}

	
	
}
